game/plugin/skills/herblore/src/potion.kt

```diff
@@ -0,0 +1,130 @@
+import org.apollo.game.action.ActionBlock
+import org.apollo.game.action.AsyncAction
+import org.apollo.game.model.Animation
+import org.apollo.game.model.entity.Player
+import org.apollo.game.plugin.api.Definitions
+import org.apollo.game.plugin.api.herblore
+import java.util.*
+
+private val MIXING_ANIM = Animation(363)
+const val WATER_VIAL = 227
+
+enum class UnfinishedPotion(
+        val id: Int,
+        herb: Herb,
+        val level: Int
+) {
+
+    GUAM(id = 91, herb = Herb.GUAM_LEAF, level = 1),
+    MARRENTILL(id = 93, herb = Herb.MARRENTILL, level = 5),
+    TARROMIN(id = 95, herb = Herb.TARROMIN, level = 12),
+    HARRALANDER(id = 97, herb = Herb.HARRALANDER, level = 22),
+    RANARR(id = 99, herb = Herb.RANARR, level = 30),
+    TOADFLAX(id = 3002, herb = Herb.TOADFLAX, level = 34),
+    IRIT(id = 101, herb = Herb.IRIT_LEAF, level = 45),
+    AVANTOE(id = 103, herb = Herb.AVANTOE, level = 50),
+    KWUARM(id = 105, herb = Herb.KWUARM, level = 55),
+    SNAPDRAGON(id = 3004, herb = Herb.SNAPDRAGON, level = 63),
+    CADANTINE(id = 107, herb = Herb.CADANTINE, level = 66),
+    LANTADYME(id = 2483, herb = Herb.LANTADYME, level = 69),
+    DWARF_WEED(id = 109, herb = Herb.DWARF_WEED, level = 72),
+    TORSTOL(id = 111, herb = Herb.TORSTOL, level = 78);
+
+    val herb = herb.identified
+
+    companion object {
+        private val UNFINISHED_POTION = UnfinishedPotion.values().associateBy { it -> it.herb }
+        operator fun get(id: Int) = UNFINISHED_POTION[id]
+    }
+}
+
+class MakeUnFinishedPotionAction(val player: Player, val unfinished: UnfinishedPotion) : AsyncAction<Player>(1, false, player) {
+    override fun action(): ActionBlock = {
+        val level = mob.herblore.current
+
+        if (level < unfinished.level) {
+            player.sendMessage("You need an herblore level of ${unfinished.level} to make this unfinished potion.")
+            stop()
+        }
+
+        if (player.inventory.remove(WATER_VIAL)) {
+            if (player.inventory.remove(unfinished.herb)) {
+                player.inventory.add(unfinished.id)
+                player.sendMessage("You have successfully made the ${Definitions.item(unfinished.id)!!.name.toLowerCase()}.")
```

Comment:
This message should be `"You mix the ${Definitions.item(unfinished.id)!!.name.toLowerCase()}. into your potion."`